util/httputil/compression.go

```diff
@@ -30,51 +31,32 @@ const (
 
 // Wrapper around http.Handler which adds suitable response compression based
 // on the client's Accept-Encoding headers.
-type compressedResponseWriter struct {
+type deflatedResponseWriter struct {
 	http.ResponseWriter
 	writer io.Writer
 }
 
 // Writes HTTP response content data.
-func (c *compressedResponseWriter) Write(p []byte) (int, error) {
+func (c *deflatedResponseWriter) Write(p []byte) (int, error) {
 	return c.writer.Write(p)
 }
 
-// Closes the compressedResponseWriter and ensures to flush all data before.
-func (c *compressedResponseWriter) Close() {
+// Closes the deflatedResponseWriter and ensures to flush all data before.
+func (c *deflatedResponseWriter) Close() {
 	if zlibWriter, ok := c.writer.(*zlib.Writer); ok {
 		zlibWriter.Flush()
 	}
-	if gzipWriter, ok := c.writer.(*gzip.Writer); ok {
-		gzipWriter.Flush()
-	}
 	if closer, ok := c.writer.(io.Closer); ok {
 		defer closer.Close()
 	}
 }
 
-// Constructs a new compressedResponseWriter based on client request headers.
-func newCompressedResponseWriter(writer http.ResponseWriter, req *http.Request) *compressedResponseWriter {
-	encodings := strings.Split(req.Header.Get(acceptEncodingHeader), ",")
-	for _, encoding := range encodings {
-		switch strings.TrimSpace(encoding) {
-		case gzipEncoding:
-			writer.Header().Set(contentEncodingHeader, gzipEncoding)
-			return &compressedResponseWriter{
-				ResponseWriter: writer,
-				writer:         gzip.NewWriter(writer),
-			}
-		case deflateEncoding:
-			writer.Header().Set(contentEncodingHeader, deflateEncoding)
-			return &compressedResponseWriter{
-				ResponseWriter: writer,
-				writer:         zlib.NewWriter(writer),
-			}
-		}
-	}
-	return &compressedResponseWriter{
+// Constructs a new deflatedResponseWriter based on client request headers.
+func newDeflateResponseWriter(writer http.ResponseWriter) *deflatedResponseWriter {
+	writer.Header().Set(contentEncodingHeader, deflateEncoding)
```

Comment:
This side-effect seems out of place in what is otherwise a constructor.

Reply:
ok.. moved to inside the switch. Is it better?